xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -29,9 +29,12 @@ using namespace ActiveAE;
 
 using namespace std::chrono_literals;
 
-#define MAX_CACHE_LEVEL 0.4   // total cache time of stream in seconds
-#define MAX_WATER_LEVEL 0.2   // buffered time after stream stages in seconds
-#define MAX_BUFFER_TIME 0.1   // max time of a buffer in seconds
+namespace
+{
+constexpr auto MAX_CACHE_LEVEL = 0.4f; // total cache time of stream in seconds;
+constexpr auto MAX_WATER_LEVEL = 0.2f; // buffered time after stream stages in seconds;
+constexpr auto MAX_BUFFER_TIME = 0.1; // max time of a buffer in seconds;
```

Comment:
can we avoid abuse of `auto` expression?
when there are many variables is better readable when types are explicit set

Reply:
I would also use explicit float here - especially in combination with the forced type at the end.
Another idea to care for rounding errors is using double if that is needed.

Reply:
> Another idea to care for rounding errors is using double if that is needed.

Use double for MAX_WATER_LEVEL constant and level calculation does not change things. Exact same numbers obtained.

Reply:
Jep - I also think that a little epsilon helps, especially if the delta is below one packet - it is clearly a rounding error somewhere in the middle.